shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/ManagedProcessorSyncServiceImpl.java

```diff
@@ -0,0 +1,96 @@
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.redhat.service.smartevents.infra.v2.api.models.ConditionStatus;
+import com.redhat.service.smartevents.infra.v2.api.models.OperationType;
+import com.redhat.service.smartevents.infra.v2.api.models.dto.ConditionDTO;
+import com.redhat.service.smartevents.infra.v2.api.models.dto.ProcessorDTO;
+import com.redhat.service.smartevents.infra.v2.api.models.dto.ProcessorStatusDTO;
+import com.redhat.service.smartevents.shard.operator.v2.converters.ProcessorStatusConverter;
+import com.redhat.service.smartevents.shard.operator.v2.resources.ManagedProcessor;
+
+import static com.redhat.service.smartevents.infra.v2.api.models.DefaultConditions.DP_PROCESSOR_DELETED_NAME;
+
+@ApplicationScoped
+public class ManagedProcessorSyncServiceImpl implements ManagedProcessorSyncService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ManagedProcessorSyncServiceImpl.class);
+
+    @Inject
+    ManagerClient managerClient;
+
+    @Inject
+    ManagedProcessorService managedProcessorService;
+
+    @Override
+    public void syncManagedProcessorWithManager() {
+        managerClient.fetchProcessorsForDataPlane()
+                .onItem()
+                .invoke(this::processDelta)
+                .subscribe().with(
+                        success -> LOGGER.debug("Successfully processed ManagedProcessor deltas"),
+                        error -> LOGGER.debug("Failed to process ManagedProcessor deltas"));
+    }
+
+    @Override
+    public void syncManagedProcessorStatusBackToManager() {
+        managerClient.fetchProcessorsForDataPlane()
+                .onItem()
+                .transform(this::transformToProcessorStatus)
+                .invoke(this::notifyProcessorStatus)
+                .subscribe().with(
+                        success -> LOGGER.debug("Successfully sync ManagedProcessor status with Manager"),
+                        error -> LOGGER.error("Failed to sync ManagedProcessor status with Manager", error));
+    }
+
+    private void processDelta(List<ProcessorDTO> processorDTOList) {
+        for (ProcessorDTO processorDTO : processorDTOList) {
+            if (processorDTO.getOperationType() == OperationType.DELETE) {
+                managedProcessorService.deleteManagedProcessor(processorDTO);
+            } else {
+                managedProcessorService.createManagedProcessor(processorDTO);
+            }
+        }
+    }
+
+    private List<ProcessorStatusDTO> transformToProcessorStatus(List<ProcessorDTO> processorDTOList) {
+        Map<String, ManagedProcessor> deployedManagedProcessors = managedProcessorService.fetchAllManagedProcessors()
+                .stream().collect(Collectors.toMap(m -> m.getSpec().getId(), m -> m));
+
+        List<ProcessorStatusDTO> processorStatusDTOS = new ArrayList<>(processorDTOList.size());
+        for (ProcessorDTO processorDTO : processorDTOList) {
```

Comment:
Defensive programming?

Reply:
@manstis Can you give an example of what you'd like to see here  in terms of being defensive?

Reply:
@robpblake Sure, I was referring to my comment [above](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/1453#discussion_r1060439723).
```
processorDTOList.forEach(this::handleProcessor);
...
void handleProcessor(ProcessorDTO processorDTO) {
  try {
    //Do whatever that could throw an exception, even if unexpected...
  } catch(Exception e) {
    LOGGER.info("Blah blah...");
  }
}
```
Should processing of one `ProcessorDTO` fail for whatever reason the others are still given a chance.

Reply:
please refer to my comment above.